### PR TITLE
Feature/throw errors

### PR DESCRIPTION
**Pull request by stevenhaching**
Added Throw Error Option

### CODE REVIEW

File: src/Aquarius.ONE.ClientSDK/Utilities/RestHelper.cs [hunk 1]
```diff
@@ -12,16 +12,19 @@ namespace ONE.Utilities
 {
     public class RestHelper
     {
-        public RestHelper(AuthenticationApi authentication, PlatformEnvironment environment, bool continueOnCapturedContext, bool logRestfulCalls)
+        public RestHelper(AuthenticationApi authentication, PlatformEnvironment environment, bool continueOnCapturedContext, bool logRestfulCalls, bool throwAPIErrors)
```
Comment (JonathanRobertson-AQI) [inline src/Aquarius.ONE.ClientSDK/Utilities/RestHelper.cs:15]: @stevenhaching  - do you know if teams are using this RestHelper class directly?  If so, maybe make this bool optional here.

Comment (stevenhaching) [reply src/Aquarius.ONE.ClientSDK/Utilities/RestHelper.cs:15]: You are totally right.  I will add this

Comment (stevenhaching) [reply src/Aquarius.ONE.ClientSDK/Utilities/RestHelper.cs]: Done

**Review by JonathanRobertson-AQI — APPROVED** [review]
Looks good, nice job!

File: src/Aquarius.ONE.ClientSDK/Utilities/RestHelper.cs [hunk 2]
```diff
@@ -66,6 +69,13 @@ public RestHelper(AuthenticationApi authentication, PlatformEnvironment environm
                 else
                     Event(null, new ClientApiLoggerEventArgs { File = file, EventLevel = EnumEventLevel.Warn, HttpStatusCode = response.StatusCode, ElapsedMs = elapsedMs, Module = "RestHelper", Message = $"PostRestJSONAsync Failed: {endpointURL}" });
 
+                if (_throwAPIErrors && !response.IsSuccessStatusCode)
```
Comment (joezhou-aqi) [inline src/Aquarius.ONE.ClientSDK/Utilities/RestHelper.cs:72]: With all of these repeated action, it might be better to extract it to a method.

**Pull request merged**